libuavcan/src/transport/uc_frame.cpp

```diff
@@ -159,11 +161,20 @@ bool Frame::compile(CanFrame& out_can_frame) const
 
     UAVCAN_ASSERT(payload_len_ < sizeof(static_cast<CanFrame*>(UAVCAN_NULLPTR)->data));
 
-    out_can_frame.dlc = static_cast<uint8_t>(payload_len_);
+    out_can_frame.dlc = CanFrame::dataLengthToDlc(static_cast<uint8_t>(payload_len_));
     (void)copy(payload_, payload_ + payload_len_, out_can_frame.data);
 
-    out_can_frame.data[out_can_frame.dlc] = tail;
-    out_can_frame.dlc++;
+    if (payload_len_ < 8) {
+        out_can_frame.data[payload_len_] = tail;
+        out_can_frame.dlc++;
+    } else if (payload_len_ == CanFrame::dlcToDataLength(out_can_frame.dlc)){
+        out_can_frame.dlc++;
+        out_can_frame.data[CanFrame::dlcToDataLength(out_can_frame.dlc) - 1] = tail;
```

Comment:
do we need to memset here to zero the pad bytes? Are all frames guaranteed to start with zeros in the data field? I'm concerned we could leak data